### PR TITLE
Better handling on feature_enabled

### PR DESCRIPTION
**Pull request by yakkomajuri**
Was looking at the feature flags and came across a pretty bad pattern where we use an error for path handling in logic we have full control over.

### CODE REVIEW

File: posthog/client.py [hunk 1]
```diff
@@ -330,9 +330,11 @@ def feature_enabled(self, key, distinct_id, default=False):
...
             response = default
         else:
-            try:
-                feature_flag = [flag for flag in self.feature_flags if flag["key"] == key][0]
-            except IndexError:
+            for flag in self.feature_flags:
+                if flag["key"] == key:
+                    feature_flag = flag
+                    break
+            else:
```
Comment (neilkakkar) [inline posthog/client.py:337]: Not sure if it's Github display, but this `else` seems indented incorrectly? 👀

Comment (yakkomajuri) [reply posthog/client.py:337]: Actually, this is correct, it's a Python for-else loop: https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loop

Comment (yakkomajuri) [reply posthog/client.py:337]: Else will run if break doesn't

Comment (neilkakkar) [reply posthog/client.py:337]: Ah, fascinating! Wasn't aware this was possible :o

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**